target/linux/ramips/rt305x/base-files/etc/board.d/02_network

```diff
@@ -230,6 +228,8 @@ ramips_setup_macs()
 	huawei,d105|\
 	hilink,hlk-rm04|\
 	nexaira,bc2|\
+	olimex,rt5350f-olinuxino|\
+	olimex,rt5350f-olinuxino-evb|\
```

Comment:
Hmm, don't really see why this particular "new" setup is chosen.

If factory 0x4 is empty, then the statement for ethernet in DTS should be removed in DTS, so a random address is actually generated. (For wmac, it will still use whatever is written in caldata?)

For 02_network, if there is no predefined MAC address, I do not see why we should change the lan_mac and assign a different wan_mac as well. I'd just have removed the devices here ...?

Reply:
A random address (for ethernet) is actually generated, as the mac address at 0x4 is "ff:ff:ff:ff:ff:ff", and the kernel will not accept that. From what I've observed, the address generated by the kernel is always local, but we'd better play safe here by forcing local bit set.
By including the device in that case, the random address generated in the first boot will be saved in /etc/config/network and will be used in subsequent boots. For wmac, another random address is generated, but it is not saved in /etc/config/wireless so it will change on every reboot.
I thought the dts statement will have no harm, but I'll remove it if you like.

Reply:
Hmm, I see why it works and what's the reason behind the 02_network entries now.

However, I don't really like the concept of relying on the calibration data to be all fs. An alternative option would be to set the local-mac-address to 00:00:00:00:00:00 or ff:ff:ff:ff:ff:ff manually in DTS, and thus trigger a random MAC address in any case.

And I'd be curious what the vendor did there and whether having a WAN port with separate address is actually justified.

Anyway, I admit this is all nitpicking, and your patch should address the problem properly as is....

Reply:
> An alternative option would be to set the local-mac-address to 00:00:00:00:00:00 or ff:ff:ff:ff:ff:ff manually in DTS

That's a nice idea. One of the reason why I left the 0x4 statement in the dts was that users might write their mac address in factory partition for some reason. But probably we don't need to care that situation.

I don't see any particular reason to have a separate wan address here, but I prefer it to writing a new case only for this device. That's all.

Reply:
It seems wmac address is not overridden by local-mac-address (nor mac-address). I'm not fond of using local-mac-address only for ethernet, so I'll just keep the current state.